internal/application/application.go

```diff
@@ -250,18 +250,7 @@ func deleteStagePVC(ctx context.Context, cluster *kubernetes.Cluster, appRef mod
 // happened, yet there is no workload. Ee.g. by calling the "staging" endpoint but not calling the "deploy"
 // endpoint. Since our client doesn't support that scenario, this method doesn't support it either.
 func PreviousStageID(ctx context.Context, cluster *kubernetes.Cluster, appRef models.AppRef) (string, error) {
-	wl := NewWorkload(cluster, appRef)
-
-	deployment, err := wl.Deployment(ctx)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return "", err
-		}
-		// App is inactive, no deployment, no workload, no id
-		return "", nil
-	}
-
-	return wl.GetStageID(ctx, deployment), nil
+	return NewWorkload(cluster, appRef).GetStageID(ctx)
```

Comment:
The only thing that makes this stage ID "the previous one" is the fact that the caller gets it before we stage again. The method itself simply returns the stage ID of the currently running app. It's up to the caller to decide when to call this. I think this method should be called just `StageID` . The caller can store it in a variable named "previousStageID" or something that makes sense.

Reply:
True.

Reply:
commit 9d3a5f190cfa088f86e69e91296b81e23833fe06